src/Notepad4.cpp

```diff
@@ -6428,57 +6489,41 @@ bool CheckIniFile(LPWSTR lpszFile, LPCWSTR lpszModule) noexcept {
 
 	if (PathIsRelative(tchFileExpanded)) {
 		WCHAR tchBuild[MAX_PATH];
-		// program directory
+		
+		// First check %LOCALAPPDATA%\Notepad4
+		LPWSTR pszPath = nullptr;
+		if (S_OK == SHGetKnownFolderPath(FOLDERID_LocalAppData, KF_FLAG_DEFAULT, nullptr, &pszPath)) {
+			PathCombine(tchBuild, pszPath, WC_NOTEPAD4);
+			CoTaskMemFree(pszPath);
+			PathAppend(tchBuild, tchFileExpanded);
+			if (PathIsFile(tchBuild)) {
+				lstrcpy(lpszFile, tchBuild);
+				return true;
+			}
+		}
+
+		// Then check program directory
 		lstrcpy(tchBuild, lpszModule);
 		lstrcpy(PathFindFileName(tchBuild), tchFileExpanded);
 		if (PathIsFile(tchBuild)) {
-			lstrcpy(lpszFile, tchBuild);
-			return true;
-		}
-
-#if _WIN32_WINNT >= _WIN32_WINNT_VISTA

```

Comment:
Old code (searches ini) could be kept as is.

Reply:
Restored support for APPDATA and PROFILE since people may already depend on that.